controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -97,12 +97,12 @@ var _ = Describe("KubeadmControlPlaneReconciler", func() {
 			Expect(result).To(Equal(ctrl.Result{}))
 
 			By("Calling reconcile should return error")
-			Expect(k8sClient.Delete(context.Background(), cluster)).To(Succeed())
+			Expect(testEnv.Delete(context.Background(), cluster)).To(Succeed())
 
-			result, err = r.Reconcile(ctrl.Request{NamespacedName: util.ObjectKey(kcp)})
-
-			Expect(err).To(HaveOccurred())
-			Expect(result.Requeue).To(BeFalse())
+			Eventually(func() error {
+				_, err := r.Reconcile(ctrl.Request{NamespacedName: util.ObjectKey(kcp)})
+				return err
+			}, 10*time.Second).Should(HaveOccurred())
```

Comment:
Why the switch to eventually here? Would it be better to have an eventually on the deletion being fully completed and then retrying the reconcile call instead?